src/main/java/org/xbill/DNS/SimpleResolver.java

```diff
@@ -339,82 +341,80 @@ private int maxUDPSize(Message query) {
         address.getPort());
     log.trace("Query:\n{}", query);
 
-    CompletableFuture<byte[]> result;
+    CompletableFuture<?> result;
     if (tcp) {
       result = NioTcpClient.sendrecv(localAddress, address, query, out, timeoutValue);
```

Comment:
I didn't want to rewrite the NioTcpClient, since you can't multicast over TCP. I was going for minimal impact to the existing code. Avahi-daemons don't listed on TCP port 5353.

Reply:
I'd rather have existing code touched a bit rather than this unsafe typecasting.
If you prefer keeping the byte[] return, then an alternative is to keep sendrecv in NioUdpClient as it was originally (returning a byte[] for unicast), and having a `sendReceiveMulticast` for mDNS returning a `List<byte[]>`. Then assign the result to different variables to avoid doing instanceof checks.